zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/AQLGeneratorTest.scala

```diff
@@ -530,6 +530,123 @@ class AQLGeneratorTest extends Specification {
+          |return {'population' : $l0.population, 'stateID' : $l0.stateId}}
+        """.stripMargin.trim
+      )
+    }
+
+    "translate lookup multiple table with one join key on each" in {
+      val populationDataSet = PopulationDataStore.DatasetName
+      val populationSchema = PopulationDataStore.PopulationSchema
+
+      val literacyDataSet = LiteracyDataStore.DatasetName
+      val literacySchema = LiteracyDataStore.LiteracySchema
+
+      val selectValuesPopulation = Seq("population")
+      val lookupPopulation = LookupStatement(Seq("geo_tag.stateID"), populationDataSet, Seq("stateId"), selectValuesPopulation,
+        selectValuesPopulation)
+      val selectValuesLiteracy = Seq("literacy")
+      val lookupLiteracy = LookupStatement(Seq("geo_tag.stateID"), literacyDataSet, Seq("stateId"), selectValuesLiteracy,
+        selectValuesLiteracy)
+
+      val selectValues = Seq("*", populationDataSet, literacyDataSet)
+      val selectStatement = SelectStatement(Seq.empty, 0, 0, selectValues)
+      val filter = Seq(textFilter)
+      val query = new Query(TwitterDataSet, Seq(lookupPopulation, lookupLiteracy), filter, Seq.empty, select = Some(selectStatement))
+      val result = parser.generate(query, Map(TwitterDataSet -> schema, populationDataSet -> populationSchema,
+        literacyDataSet -> literacySchema))
+      removeEmptyLine(result) must_== unifyNewLine(
+        """
+          |for $t in dataset twitter.ds_tweet
+          |where similarity-jaccard(word-tokens($t.'text'), word-tokens('zika')) > 0.0
+          |and contains($t.'text', "virus")
+          |limit 0
+          |offset 0
+          |return
+          |{ '*': $t, 'twitter.US_population': for $l0 in dataset twitter.US_population
+          |where $t.'geo_tag'.'stateID' /* +indexnl */ = $l0.stateId
+          |return {'population' : $l0.population}, 'twitter.US_literacy': for $l1 in dataset twitter.US_literacy
+          |where $t.'geo_tag'.'stateID' /* +indexnl */ = $l1.stateId
+          |return {'literacy' : $l1.literacy}}
+        """.stripMargin.trim
+      )
+    }
+
+
+    "translate group by query having lookup with one join key" in {
+      val populationDataSet = PopulationDataStore.DatasetName
+      val populationSchema = PopulationDataStore.PopulationSchema
+
+      val selectValues = Seq("population")
+      val group = GroupStatement(Seq(byState), Seq(AggregateStatement("population", Sum, "sum")))
+      val lookup = LookupStatement(Seq("geo_tag.stateID"), populationDataSet, Seq("stateId"), selectValues,
+        selectValues)
+      val filter = Seq(textFilter)
+      val query = new Query(TwitterDataSet, Seq(lookup), filter, Seq.empty, groups = Some(group))
+      val result = parser.generate(query, Map(TwitterDataSet -> schema, populationDataSet -> populationSchema))
+      removeEmptyLine(result) must_== unifyNewLine(
+        """
+          |for $t in dataset twitter.ds_tweet
+          |for $l0 in dataset twitter.US_population
```

Comment:
As a baseline implementation for the lookup is that we compose an augemented dataset by append new columes, and then operate the rest of the statements (e.g., select, groupby)

```
for $t in dataset twitter.ds_tweetwhere
where similarity-jaccard(word-tokens($t.'text'), word-tokens('zika')) > 0.0
and contains($t.'text', "virus")
return {
'creat_at' : $t.create_at
'id' : $t.id
....  /// **flatten all $t's field by reading them from the schema.** 
'population' : ( for $l0 in dataset twitter.US_population 
                         where $l0.stateID = $t.'geo_tag'.'stateID'
                         return $l0.population) [0]  /// **here since it's lookup, let assume**
                                                                     ///  **if it gonna hit multiple values,** 
                                                                     /// **they should be same, then we just pick one.**
'other-lookups' : (xxxx)[0]
}

/// **then we can use the same logic as if we are group on a single table.** 
let $l0aggr := $l0.population
group by $g0 := $t.geo_tag.stateID with $l0aggr
return {
  'state' : $g0,'sum' : sum($l0aggr)
}
```
Above example is not guaranteed to be optimal. But let's first make a working one 🥇